src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -584,6 +590,11 @@ protected override void ComputeDependencyNodeDependencies(List<DependencyNodeCor
                             corInfoImpl.CompileMethod(methodCodeNodeNeedingCode, Logger);
                         }
                     }
+                    catch (Internal.TypeSystem.TypeSystemException.TypeLoadException)
```

Comment:
I agree with Michal here. We cannot catch `TypeLoadException` and assume it is associated with the `GenericCycleDetector`. You either need to wrap the calls to `GenericCycleDetector` in a try/catch and translate the exception to something well known, in order to print this out, or modify the `GenericCycleDetector` to produce a unique failure that can be identified.

Reply:
Removed special handling per Michal's PR feedback in the latest commit.